lorawan-device/src/mac/session.rs

```diff
@@ -382,11 +386,20 @@ impl Session {
                         configuration.rx2_frequency = Some(freq);
                     }
 
+                    // RXParamSetupReq has its own acknowledgment mechanism, requiring
+                    // RXParamSetupAns with all uplinks until a Class A downlink is received
+                    // by the end-device.
                     let mut cmd = RXParamSetupAnsCreator::new();
                     cmd.set_rx1_data_rate_offset_ack(true)
```

Comment:
are we always acking the dr offset? Is there no check for this?

Reply:
Yes, unfortunately this is how it has been from the "start".
I'll be working on these two once the acknowledgment mechanism is in place:
* RX2 DR - quite trivial to implement
* RX1 DR offset - somewhat more tricky as we have a lot more regions, but I have already an idea fleshed out based on feedback from #354

Reply:
ok, step by step makes sense. I am ok with merging stuff that doesn't make it worse or break anything that was working